agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMContextImplEventRunnable.java

```diff
@@ -38,6 +38,7 @@ public void run() {
                 rmContext.getRMApps().forEach((applicationId, rmApp) -> {
                     if (cacheFinishedApp.getIfPresent(applicationId.toString()) == null) {
                         Header header = Header.newBuilder()
+                                .withId(applicationId.toString())
```

Comment:
Id = ApplicationID, is this expected?

Reply:
Yes it is if field is only used as key for kafka partitioning
in case of app it is app id, in case of other is is currently HeaderUtils.getId